app/services/course/assessment/submission/auto_grading_service.rb

```diff
@@ -38,6 +50,13 @@ def grade_answers(submission)
   def grade_answer(answer)
     raise ArgumentError if answer.changed?
     answer.auto_grade!
+    # Catch errors if answer is in attempting state, caused by a race condition where
+    # a new attempting answer is created while the submission is finalised, but before the
+    # autograding job is executed.
+  rescue IllegalStateError
```

Comment:
> so that new developers won't accidentally call #auto_grade! on attempting answers.
Agree with this, actually even ourselves might manually call `auto_grade!` to fix some db issues sometimes, it's better to raise an error if the answer is not in the correct state and `submitted_at` is not present.

It doesn't make sense to let `auto_grade` set the `submitted_at`, or we should add a new method.